src/generators/gencs.ml

```diff
@@ -3107,7 +3107,7 @@ let generate con =
 
 		ExpressionUnwrap.configure gen;
 
-		UnnecessaryCastsRemoval.configure gen;
+		(* UnnecessaryCastsRemoval.configure gen; *)
```

Comment:
Will this result in even more casts generated? O_o

Reply:
That filter only takes of block-level casts which is not a sound thing to do. Maybe it should be revised to actually check for unnecessary casts.